src/v/archival/ntp_archiver_service.cc

```diff
@@ -508,6 +508,48 @@ ss::future<> ntp_archiver::sync_manifest_until_term_change() {
     }
 }
 
+ss::future<> ntp_archiver::unsafe_reset_metadata(ss::input_stream<char> is) {
```

Comment:
We might want to also replicate `update_start_kafka_offset_cmd` and `truncate_archive_commit_cmd` too in order to set the `_start_kafka_offset` and `_archive_clean_offset` to the values from the deserialised manifest. The current approach will work Redpanda <= v23.1 though, but it will become incorrect on v23.2

Reply:
I had the same thought, but I opted to add this later, and leave the initial escape hatch as something backportable.